### DISCUSSION TITLE
Client manual upgrade

### DISCUSSION THREAD

**Discussion by MapInetum** [Mimetis/Dotmim.Sync#857] [Q&A] [answered]
Hello.
Currently, we have version 0.9.5 installed.
We see the need to update to version 0.9.6 but we have encountered the problem that when updating the version it shows us the following error message:
"This version needs a client manual upgrade"
We have a large volume of synchronized data and we don't see it feasible to have to recreate all the scopes again.
Is there a way to migrate without having to recreate everything again?
Thanks.

**Comment by MapInetum** [comment]
Hi. I find the 'solution' for me.
https://github.com/Mimetis/Dotmim.Sync/discussions/802

Thanks for all.

**Reply by Mimetis** [reply]
Not too complicated ?

**Reply by MapInetum** [reply]
I have been trying to migrate all morning and indeed, it is complicated and I receive some other exception when executing it.
Any alternative without having to sync from scratch?

**Reply by Mimetis** [reply]
https://github.com/Mimetis/Dotmim.Sync/discussions/802#discussioncomment-3594681

**Reply by MapInetum** [reply]
Hi @Mimetis, i find a problem with this solution in my case.
We sync a various database clients in the same server with differents scope names, but the utilizing filters are the sames.

Differents scopes with the same filters
![image](https://user-images.githubusercontent.com/79869950/200579975-22d1a751-8944-4fb4-b19d-3bf96775aeaa.png)

Exception in the UpgradeAsync() method
![image](https://user-images.githubusercontent.com/79869950/200580195-12c13949-3950-4774-bfc7-56d9e2337186.png)

I find the thows in the SetupFilters.cs
![image](https://user-images.githubusercontent.com/79869950/200580368-9d5ef5d9-85e1-4037-ab34-84062d3d1114.png)

I solve this with this code:

_public void Add(SetupFilter item)
{
    if (!InnerCollection.Any(st => item.EqualsByName(st)))
    {
        this.InnerCollection.Add(item);
    }
}_

What do you think?

**Reply by Mimetis** [reply]
well, the thing is that we should never hit the point where you are adding a filter two times.
Can you share a small sample where you are reproducing the error ?

**Comment by camiteca** [comment, marked as answer]
Hi @Mimetis, the use case for @maprado-inetum is a central database being synchronized to N child databases. The filter is actually the same, but the data being synchronized is different in each "child" database, having only a small subset of the central one.

**Reply by MapInetum** [reply]
![image](https://user-images.githubusercontent.com/79869950/200588097-a48b68e5-8e48-4939-9631-1420f1e6813d.png)
